### PR TITLE
Jl/license

### PR DESCRIPTION
**Pull request by lav-julien**
## Issue

Add the different license items to the rewrite repository so it gets merged at the same time than the rest of the rewrite

## Description

Actions to perform are:

- [x] Add a License.txt file for GNU-AGPL
- [x] Add a license badge at the top of the readme file
- [x] Add a license section in the readme file linking to the license.txt file
- [x] Add the workflow for contributions review in the readme file
- [x] Link the contribution guidelines file from Viktor in the readme file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,10 @@ Join us on [Slack](https://slack.electricitymaps.com) if you wish to discuss dev
 
 We would love your feedback on how to improve the contribution experience, please feel free to fill out this [form](https://forms.gle/VRWvEFwhtnhpzPVX8)
 
+## License
+
+Electricity Maps is licensed under GNU-AGPL. Find our license [here](link to license file)
```
Comment (VIKTORVAV99) [inline README.md:29]:
```suggestion
Electricity Maps is licensed under GNU-AGPLv3. Find our license [here](link to license file)
```
Should we include the licence version as well?
(This is the identifier used by [choosealicense](https://choosealicense.com/licenses/agpl-3.0/))

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Electricity Maps
 
-[![Slack Status](https://slack.electricitymaps.com/badge.svg)](https://slack.electricitymaps.com) [![Twitter Follow](https://img.shields.io/twitter/follow/ElectricityMaps.svg?style=social&label=Follow)](https://twitter.com/ElectricityMaps)
+<img src="https://img.shields.io/github/license/electricitymaps/electricitymaps-contrib-rewrite" alt="Electricity Maps is released under the GNU-AGPL license." /> [![Slack Status](https://slack.electricitymaps.com/badge.svg)](https://slack.electricitymaps.com) [![Twitter Follow](https://img.shields.io/twitter/follow/ElectricityMaps.svg?style=social&label=Follow)](https://twitter.com/ElectricityMaps)
```
Comment (VIKTORVAV99) [inline README.md:3]:
```suggestion
<img src="https://img.shields.io/github/license/electricitymaps/electricitymaps-contrib-rewrite" alt="Electricity Maps is released under the GNU-AGPLv3 license." /> [![Slack Status](https://slack.electricitymaps.com/badge.svg)](https://slack.electricitymaps.com) [![Twitter Follow](https://img.shields.io/twitter/follow/ElectricityMaps.svg?style=social&label=Follow)](https://twitter.com/ElectricityMaps)
```
Same suggestion here as above.

**Review by VIKTORVAV99 — COMMENTED** [review]
Just one tiny suggestion, nice choice of license though!

**Comment by VIKTORVAV99** [comment]
We should probably also update the license information in the `package.json` files (web, mobile and mockserver) and the `pyproject.toml` file. Even though we don't publish these packages I think it would be good for clarity and to ensure there are no conflicting licenses in the repo.

**Comment by lav-julien** [comment]
> We should probably also update the license information in the `package.json` files (web, mobile and mockserver) and the `pyproject.toml` file. Even though we don't publish these packages I think it would be good for clarity and to ensure there are no conflicting licenses in the repo.

**Comment by lav-julien** [comment]
Link not working in the Readme but I guess it's because the PR is not merged, don't know if there is a more robust way to handle this in GitHub

**Comment by VIKTORVAV99** [comment]
> Link not working in the Readme but I guess it's because the PR is not merged, don't know if there is a more robust way to handle this in GitHub

It's also pointing here to the rewrite repository, once it's merged to master it will break again unfortunately.

I would probably link it to the license file in the "normal" repository so it works when we merge it to master.

(In either case we have a lot of these small cleanups to do once we merge the repositories).

**Comment by lav-julien** [comment]
@VIKTORVAV99 do you think we can merge this pull request in the contrib-rewrite and then we can start doing some of these cleanup things? Or what would be the best way to proceed?

Also, for the workflow diagram, I would suggest to upload an image on our repository and then embed it in the README file, let me know what you think about it.

workflow
I would suggest this workflow to keep something simple and easy to understand

**Comment by VIKTORVAV99** [comment]
> @VIKTORVAV99 do you think we can merge this pull request in the contrib-rewrite and then we can start doing some of these cleanup things? Or what would be the best way to proceed?
> 
> Also, for the workflow diagram, I would suggest to upload an image on our repository and then embed it in the README file, let me know what you think about it.
> 
> workflow
> I would suggest this workflow to keep something simple and easy to understand

Yeah I think this can be merged but someone, @madsnedergaard, forgot to give me write permissions in the fork. 😅

So either Tony or Mads will have to approve it and merge it...

As for the workflow diagram I have been looking into if we can use https://mermaid.js.org/, it should be fully supported by github but I'll have to look up if it works in mobile app too. Otherwise I think embedding the image is a good idea.

**Comment by VIKTORVAV99** [comment]
> @VIKTORVAV99 do you think we can merge this pull request in the contrib-rewrite and then we can start doing some of these cleanup things? Or what would be the best way to proceed?
> 
> Also, for the workflow diagram, I would suggest to upload an image on our repository and then embed it in the README file, let me know what you think about it.
> 
> workflow
> 
> I would suggest this workflow to keep something simple and easy to understand

I have been thinking a little more about the diagram itself and I would probably change the first check, valid code formatting, to something like: "Passes CI Checks". This is because we are not just checking the code formatting but also running unit tests, component test (soon) and full E2E tests not to mention scanning for vulnerabilities so it's a little more than just formatting.

**Comment by VIKTORVAV99** [comment]
If we use mermaid it would look something like this:
```mermaid
---
title: PR workflow
---
flowchart LR
    PR(New PR)
    CI{Passes CI checks?}
    CIFAIL(Reject)
    REVIEW(Review by Electricity Maps team)
    REVIEWAPPROVED{Approving review?}
    APPROVED(Approved)
    REQUESTCHANGES(Request changes)
    MERGE((Merge))

    PR --> CI
    CI -- YES --> REVIEW
    CI -- NO --> CIFAIL
    REVIEW --> REVIEWAPPROVED
    REVIEWAPPROVED -- YES --> APPROVED
    REVIEWAPPROVED -- NO --> REQUESTCHANGES
    REQUESTCHANGES --> REVIEW
    APPROVED --> MERGE
```
The benefits of this is that we can make it interactive and potentially link to more resources if needed.

**Comment by madsnedergaard** [comment]
I just added the license to mobile and pyproject files. I also changed it to use `AGPL-3.0-or-later`, which is the "official" identifier (see https://spdx.org/licenses/) :)

Regarding the diagram I think we should just use an image for now to keep it simple, but I prefer @VIKTORVAV99's phrasing of the first step. I also think "reject" is a bit too harsh, maybe we can say "request changes" and point it back to the CI check or something similar?

**Comment by VIKTORVAV99** [comment]
> I just added the license to mobile and pyproject files. I also changed it to use `AGPL-3.0-or-later`, which is the "official" identifier (see https://spdx.org/licenses/) :)
> 
> Regarding the diagram I think we should just use an image for now to keep it simple, but I prefer @VIKTORVAV99's phrasing of the first step. I also think "reject" is a bit too harsh, maybe we can say "request changes" and point it back to the CI check or something similar?

Maybe just make changes instead of request changes?

Image will probably be best too, mermaid don't really seem to work on mobile as it should.

**Comment by lav-julien** [comment]
> > I just added the license to mobile and pyproject files. I also changed it to use `AGPL-3.0-or-later`, which is the "official" identifier (see https://spdx.org/licenses/) :)
> > Regarding the diagram I think we should just use an image for now to keep it simple, but I prefer @VIKTORVAV99's phrasing of the first step. I also think "reject" is a bit too harsh, maybe we can say "request changes" and point it back to the CI check or something similar?
> 
> Maybe just make changes instead of request changes?
> 
> Image will probably be best too, mermaid don't really seem to work on mobile as it should.

I uploaded a new workflow in the main path (let me know if we should store it in a specific folder).
EMaps contrib workflow

I think we could merge the PR now and start fixing the links. Let me know @VIKTORVAV99 if you agree

**Comment by VIKTORVAV99** [comment]
> > > I just added the license to mobile and pyproject files. I also changed it to use `AGPL-3.0-or-later`, which is the "official" identifier (see https://spdx.org/licenses/) :)
> > > Regarding the diagram I think we should just use an image for now to keep it simple, but I prefer @VIKTORVAV99's phrasing of the first step. I also think "reject" is a bit too harsh, maybe we can say "request changes" and point it back to the CI check or something similar?
> > 
> > Maybe just make changes instead of request changes?
> > 
> > Image will probably be best too, mermaid don't really seem to work on mobile as it should.
> 
> I uploaded a new workflow in the main path (let me know if we should store it in a specific folder).
> EMaps contrib workflow
> 
> I think we could merge the PR now and start fixing the links. Let me know @VIKTORVAV99 if you agree

If we add more assets it might make sense to create a specific folder for it (something like "doc assets" maybe.

I think we can merge this now unless @madsnedergaard has anything he would like to add.

Then I'll update the contribution guide document to reference the new license as well. (BTW we will probably cut out most of the existing contribution guide in the open PR and put it in the wiki step by step later).

**Review by madsnedergaard — APPROVED** [review]
Hurray, amazing work! Thanks both of you 🥳

**Pull request merged**